artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -508,6 +524,9 @@ public WireFormat getMarshaller() {
    }
 
    private void shutdown(boolean fail) {
+
+      this.ttlLatch.countDown();
```

Comment:
If you didn't use a thread you wouldn't need this.

just use the scheduled executor to send pings to the client please.

Cancel the executor from a future.